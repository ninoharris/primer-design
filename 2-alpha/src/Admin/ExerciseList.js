import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import moment from 'moment'
import { exercisesByIdSelector } from '../selectors/index'
import { getFilteredSortedExercises } from '../selectors/admin'

export class ExerciseList extends Component {
  renderExerciseItem = (exercise) => {
    return (
      <tr key={exercise.id}>
        <td>
          {exercise.questionPart1}
          <small className="ID">{exercise.id}</small>
        </td>
        <td className="small">{moment(exercise.createdAt).format("ddd, Do MMM YY")}</td>
        <td className="small">{moment(exercise.lastModified).format("ddd, Do MMM YY")}</td>
        <td>{exercise.authorId}</td>
        <td>
          <Link to={`/admin/edit/${exercise.id}`}>
            <button className="btn btn-primary">Edit exercise</button>
          </Link>
        </td>
      </tr>
    )
  }
  render() {
    const { exercisesList = [] } = this.props 
    return (
      <table className="table Admin-Exercises-List">
        <thead>
          <tr>
            <th>Exercise name</th>
            <th>Date added</th>
            <th>Last modified</th>
            <th>Author ID</th>
            <th></th>
          </tr>
        </thead>
        <tbody>
          {exercisesList.length > 0 ? 
            exercisesList.map(exercise => this.renderExerciseItem(exercise)) :
            <tr><td>No exercises found...</td></tr>
          }
        </tbody>
      </table>
    )
  }
}

ExerciseList.propTypes = {
  exercisesList: PropTypes.arrayOf(PropTypes.object)
}

const mapStateToProps = (state) => ({
  exercisesList: getFilteredSortedExercises(state),
})

export default connect(mapStateToProps)(ExerciseList)