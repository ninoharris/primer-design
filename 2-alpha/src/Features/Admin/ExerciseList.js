import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { exercisesListSelector, exercisesByIdSelector } from '../../selectors/index'
import { exercisesList, exercisesById } from '../../reducers/exercises';


class ExerciseList extends Component {
  renderExerciseItem = (exercise) => {
    return (
      <tr key={exercise.id}>
        <td>{exercise.question.part1}</td>
        <td>{new Date(exercise.createdAt).toLocaleDateString()}</td>
        <td>{new Date(exercise.lastModified).toLocaleDateString()}</td>
        <td>{exercise.authorId}</td>
        <td>
          <Link to={`/admin/edit/${exercise.id}`}>
            <button className="btn btn-primary">Edit exercise</button>
          </Link>
        </td>
      </tr>
    )
  }
  render() {
    return (
      <table className="table">
        <thead>
          <tr>
            <th>Exercise name</th>
            <th>Date added</th>
            <th>Last modified</th>
            <th>Author ID</th>
            <th></th>
          </tr>
        </thead>
        <tbody>
          {this.props.exercisesList.map(
            (ID) => this.renderExerciseItem(this.props.exercisesById[ID]))}
        </tbody>
      </table>
    )
  }
}

const mapStateToProps = (state) => ({
  exercisesById: exercisesByIdSelector(state),
  exercisesList: exercisesListSelector(state),
})

export default connect(mapStateToProps)(ExerciseList)