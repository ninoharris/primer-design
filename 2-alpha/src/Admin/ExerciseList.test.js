import _ from 'lodash'
import React from 'react'
import { shallow } from 'enzyme'
import { ExerciseList } from './ExerciseList'
import { data } from '../tests/exercises_fixtures'

describe('exerciseList', () => {
  test('should render a table row that tells the user that there are no exercises within their filters.', () => {
  const wrapper = shallow(<ExerciseList exerciseList={[]} />)
  expect(wrapper).toMatchSnapshot()
  })
  test('should render exerciseList with list of exercises', () => {
    const exercisesArray = _.flatMap(data, (val, key) => ({ ...val, id: key }))
    const wrapper = shallow(<ExerciseList exercisesList={exercisesArray} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('tbody tr').length).toBe(exercisesArray.length)
  })
})