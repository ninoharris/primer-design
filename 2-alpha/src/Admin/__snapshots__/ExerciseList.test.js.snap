// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exerciseList should render a table row that tells the user that there are no exercises within their filters. 1`] = `
<table
  className="table Admin-Exercises-List"
>
  <thead>
    <tr>
      <th>
        Exercise name
      </th>
      <th>
        Date added
      </th>
      <th>
        Last modified
      </th>
      <th>
        Author ID
      </th>
      <th />
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        No exercises found...
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`exerciseList should render exerciseList with list of exercises 1`] = `
<table
  className="table Admin-Exercises-List"
>
  <thead>
    <tr>
      <th>
        Exercise name
      </th>
      <th>
        Date added
      </th>
      <th>
        Last modified
      </th>
      <th>
        Author ID
      </th>
      <th />
    </tr>
  </thead>
  <tbody>
    <tr
      key="one"
    >
      <td>
        <small
          className="ID"
        >
          one
        </small>
      </td>
      <td
        className="small"
      >
        Thu, 1st Jan 70
      </td>
      <td
        className="small"
      >
        Thu, 1st Jan 70
      </td>
      <td>
        sedm1000
      </td>
      <td>
        <Link
          replace={false}
          to="/admin/edit/one"
        >
          <button
            className="btn btn-primary"
          >
            Edit exercise
          </button>
        </Link>
      </td>
    </tr>
    <tr
      key="two"
    >
      <td>
        <small
          className="ID"
        >
          two
        </small>
      </td>
      <td
        className="small"
      >
        Thu, 1st Jan 70
      </td>
      <td
        className="small"
      >
        Thu, 1st Jan 70
      </td>
      <td>
        some2999
      </td>
      <td>
        <Link
          replace={false}
          to="/admin/edit/two"
        >
          <button
            className="btn btn-primary"
          >
            Edit exercise
          </button>
        </Link>
      </td>
    </tr>
    <tr
      key="three"
    >
      <td>
        <small
          className="ID"
        >
          three
        </small>
      </td>
      <td
        className="small"
      >
        Thu, 1st Jan 70
      </td>
      <td
        className="small"
      >
        Thu, 1st Jan 70
      </td>
      <td>
        sedm4648
      </td>
      <td>
        <Link
          replace={false}
          to="/admin/edit/three"
        >
          <button
            className="btn btn-primary"
          >
            Edit exercise
          </button>
        </Link>
      </td>
    </tr>
  </tbody>
</table>
`;
